lib/compiler/src/compiler.rs

```diff
@@ -81,21 +81,29 @@ pub trait Compiler: Send {
         features: &Features,
         data: &'data [u8],
     ) -> Result<(), CompileError> {
-        let mut validator = Validator::new();
+        assert!(
```

Comment:
The module linking proposal has been entirely supplanted by components. We should nuke any reference to the module linking proposal entirely.